packages/discord.js/typings/index.d.ts

```diff
@@ -6581,17 +6636,17 @@ export interface ClientApplicationInstallParams {
   permissions: Readonly<PermissionsBitField>;
 }
 
-export type Serialized<T> = T extends symbol | bigint | (() => any)
+export type Serialized<Value> = Value extends symbol | bigint | (() => any)
   ? never
-  : T extends number | string | boolean | undefined
-  ? T
-  : T extends { toJSON(): infer R }
-  ? R
-  : T extends ReadonlyArray<infer V>
-  ? Serialized<V>[]
-  : T extends ReadonlyMap<unknown, unknown> | ReadonlySet<unknown>
+  : Value extends number | string | boolean | undefined
+  ? Value
+  : Value extends { toJSON(): infer Result }
```

Comment:
Rather than `Result` (which is too generic and a bit hard to understand what `Result` is in the following line), maybe `JSONResult` or `ToJSONReturnValue`?